### PR TITLE
[fix](Nereids) update immutable LogicalAggregate attribute by mistake

### PR DESCRIPTION
**Pull request by jackwener**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.
- logcialAgg exist mutable member.
- refactor UT.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.21 seconds
 load time: 569 seconds
 storage size: 17154699378 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221027234833_clickbench_pr_34910.html

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java [hunk 1]
```diff
@@ -231,13 +231,14 @@ public boolean equals(Object o) {
...
             return false;
         }
+        // TODO: reconsider children, because children is mutable.
         return children.equals(that.children) && plan.equals(that.plan)
                 && plan.getLogicalProperties().equals(that.plan.getLogicalProperties());
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(children, plan);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/memo/GroupExpression.java:240]: add children back

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 38.42 seconds
 load time: 571 seconds
 storage size: 17154810761 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028195705_clickbench_pr_35389.html

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.16 seconds
 load time: 559 seconds
 storage size: 17154644840 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028133540_clickbench_pr_35451.html

**Review by morrySnow — DISMISSED** [review]

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**